src/main/config/config_eeprom.c

```diff
@@ -147,19 +132,24 @@ bool isEEPROMContentValid(void)
             return false;
         }
 
-        chk = updateChecksum(chk, p, record->size);
+        crc = crc16_ccitt_update(crc, p, record->size);
 
         p += record->size;
     }
 
     const configFooter_t *footer = (const configFooter_t *)p;
-    chk = updateChecksum(chk, footer, sizeof(*footer));
+    crc = crc16_ccitt_update(crc, footer, sizeof(*footer));
     p += sizeof(*footer);
-    chk = ~chk;
-    const uint8_t checkSum = *p;
-    p += sizeof(checkSum);
+
+    // include stored CRC in the CRC calculation
+    const uint16_t *storedCrc = (const uint16_t *)p;
+    crc = crc16_ccitt_update(crc, storedCrc, sizeof(*storedCrc));
```

Comment:
Just a detail - `crc = crc16_ccitt_update(crc, p, sizeof(crc));  p += sizeof(crc);` is consistent with rest of code and slightly shorter ...